servant/src/Servant/API/ContentTypes.hs

```diff
@@ -219,9 +226,10 @@ instance AllMime '[] where
     allMime _ = []
 
 instance (Accept ctyp, AllMime ctyps) => AllMime (ctyp ': ctyps) where
-    allMime _ = (contentType pctyp):allMime pctyps
-      where pctyp  = Proxy :: Proxy ctyp
-            pctyps = Proxy :: Proxy ctyps
+    allMime _ = NE.toList (contentTypes pctyp) ++ allMime pctyps
+      where
+        pctyp  = Proxy :: Proxy ctyp
+        pctyps = Proxy :: Proxy ctyps
```

Comment:
I wish we would have less cryptic variable names. `pctyps` is not a good name. (I'm aware that you didn't introduce these names here.) `contentTypesProxy` (or `contentTypesP`) is better IMO.